src/initialize_dae.jl

```diff
@@ -150,15 +171,25 @@ function _initialize_dae!(integrator, prob::ODEProblem, alg::ShampineCollocation
             out .-= tmp
             nothing
         end
-        r = nlsolve(nlequation!, u0, autodiff = isad ? :forward : :central,
-                    method = :newton)
-        integrator.u .= r.zero
+
+        if alg.nlsolve !== nothing
+            nlsolve = alg.nlsolve
+        else
```

Comment:
why doesn't this have the simplenewtonrhapson case?

Reply:
Simple is only for oop problems